codegen/smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpStringValuesMapSerializerTest.kt

```diff
@@ -53,7 +53,8 @@ class HttpStringValuesMapSerializerTest {
 
     @Test
     fun `it handles primitive header shapes when different mode`() {
-        val contents = getTestContents(defaultModel, "com.test#PrimitiveShapesOperation", HttpBinding.Location.HEADER)
+        val settings = defaultModel.defaultSettings(defaultValueSerializationMode = DefaultValueSerializationMode.WHEN_DIFFERENT)
+        val contents = getTestContents(defaultModel, "com.test#PrimitiveShapesOperation", HttpBinding.Location.HEADER, settings)
```

Comment:
I still think we should have tests for the new behavior

https://github.com/smithy-lang/smithy-kotlin/pull/1302#discussion_r2159672896

Reply:
We already had/have tests which explicitly set `ALWAYS`. My changes in this PR make all of the implicit `WHEN_DIFFERENT` values into explicit ones. Are there specific tests you'd like to see added/expanded?